services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -683,6 +685,8 @@ public void handleDeletedInstances(Set<Instance> deletedInstances) {
         }
       });
     }
+    emptyPushZSTDDictionary =
+        Lazy.of(() -> ByteBuffer.wrap(ZstdWithDictCompressor.buildDictionaryOnSyntheticAvroData()));
```

Comment:
Can we also update the `buildDictionaryOnSyntheticAvroData` function so that everything within it is within a try catch, and we could log whatever happens before rethrowing?

Reply:
Good call, will do.